rs/sns/testing/src/sns.rs

```diff
@@ -32,54 +54,268 @@ pub struct TestCanisterInitArgs {
-        *application_subnet_id,
+        application_subnet_id,
         Encode!(&args).unwrap(),
         Some(test_canister_wasm),
         vec![ROOT_CANISTER_ID.get()],
     )
     .await
 }
 
-pub async fn create_sns(
+// PocketIC-specific version of 'create_sns' function.
+// Generates all required identities internally
+// Takes the list of IDs of the DApps that will be added to the SNS as an argument.
+//
+// Returns SNS canisters IDs and the dev participant ID.
+pub async fn create_sns_pocket_ic(
     pocket_ic: &PocketIc,
+    dev_participant_id: PrincipalId,
+    treasury_principal_id: PrincipalId,
     dapp_canister_ids: Vec<CanisterId>,
-) -> (Sns, ProposalId) {
-    let sns_proposal_id = "1";
-    let create_service_nervous_system = CreateServiceNervousSystemBuilder::default()
+) -> Sns {
+    let dev_participant = PocketIcAgent::new(pocket_ic, dev_participant_id);
+
+    let swap_treasury_agent = PocketIcAgent::new(pocket_ic, treasury_principal_id);
+    let swap_partipants_agents = (1..20)
+        .map(|i| PocketIcAgent::new(pocket_ic, PrincipalId::new_user_test_id(1000 + i as u64)))
+        .collect();
+    create_sns(
+        &dev_participant,
+        NNS_NEURON_ID,
+        &dev_participant,
+        &swap_treasury_agent,
+        swap_partipants_agents,
+        dapp_canister_ids,
+    )
+    .await
+}
+
+// Creates SNS using agents provided as arguments:
+// 1) neuron_agent - agent that controlls 'neuron_id'.
+// 2) neuron_id - ID of the neuron that has a sufficient amount of stake to propose the SNS creation and adopt the proposal.
+// 2) dev_participant_agent - Agent that will be used as an initial neuron in a newly created SNS. All other
+//    neurons will follow the dev neuron.
+// 3) swap_treasury_agent - Agent for the identity that has sufficient amout of ICP tokens to close the swap and
+//    pay for cycles needed canister upgrade as well as all associated costs.
+// 4) swap_participants_agents - Agents for the identities that will participate in the swap. The actual number of participants
+//    is determined by the swap parameters. So only some of these agents might be used. Each agent participates in the swap,
+//    follows the dev neuron for 'UpgradeSnsControlledCanister' proposals and increases its dissolve delay to be able to vote.
+// 5) dapp_canister_ids - Canister IDs of the DApps that will be added to the SNS.
+//
+// Returns SNS canisters IDs.
+pub async fn create_sns<C: CallCanisters + ProgressNetwork>(
+    neuron_agent: &C,
+    neuron_id: NeuronId,
+    dev_participant_agent: &C,
+    swap_treasury_agent: &C,
+    swap_participants_agents: Vec<C>,
+    dapp_canister_ids: Vec<CanisterId>,
+) -> Sns {
+    let mut create_service_nervous_system = CreateServiceNervousSystemBuilder::default()
         .neurons_fund_participation(true)
         .with_dapp_canisters(dapp_canister_ids)
         .build();
+
+    // Ugly hack to set developer identity to have initial neuron eligeble for voting
```

Comment:
I'm not sure this is a hack. Real-world SNSs also specify developer neurons in a similar way. Am I missing something?